tools/kompos/src/messages.ts

```diff
@@ -0,0 +1,207 @@
+
+export interface IdMap<T> {
+  [key: string]: T;
+}
+
+export interface Source {
+  id:         string;
+  sourceText: string;
+  mimeType:   string;
+  name:       string;
+  uri:        string;
+}
+
+export interface SimpleSourceSection {
+  firstIndex?: number;
+  length:      number;
+  line:        number;
+  column:      number;
+}
+
+export interface SourceSection extends SimpleSourceSection {
+  id:           string;
+  description?: string;
+  sourceId?:    string;
+}
+
+export interface Method {
+  name:          string;
+  definition:    SimpleSourceSection[];
+  sourceSection: SourceSection;
+}
+
+interface Frame {
+  sourceSection: SourceSection;
+  methodName: string;
+}
+
+interface TopFrame {
+  arguments: string[];
+  slots:     IdMap<string>;
+}
+
+export type Message = SourceMessage | SuspendEventMessage | MessageHistoryMessage;
+
+export interface SourceMessage {
+  type:     "source";
+  sources:  IdMap<Source>;
+  sections: IdMap<SourceSection>;
+  methods:  Method[];
+}
+
+export interface SuspendEventMessage {
+  type:     "suspendEvent";
+  sourceId: string;
+  sections: SourceSection[];
+  stack:    Frame[];
+  topFrame: TopFrame;
+  id: string;
+}
+
+export interface MessageHistoryMessage {
+  type: "messageHistory";
+  messageHistory: any; // TODO
+}
+
+export type BreakpointData = LineBreakpointData | SendBreakpointData | AsyncMethodRcvBreakpointData;
```

Comment:
Stuff is going to change around soon again. So, I put them all in here for the moment.
There is also currently rather bad choice of terms with Message/Respond. That's very depending on what you consider sender/receiver. So, doesn't make sense universally. We'll need to find better terminology there, too.